app/Http/Controllers/Auth/ForgotPasswordController.php

```diff
@@ -18,15 +24,133 @@ class ForgotPasswordController extends Controller
     |
     */
 
-    use SendsPasswordResetEmails;
+    /**
+     * @var
+     */
+    private $hashRepository;
+
+
+    /**
+     * @var
+     */
+    private $mailManager;
+
+    /**\
+     * @var
+     */
+    private $userRepository;
 
     /**
-     * Create a new controller instance.
-     *
-     * @return void
+     * ForgotPasswordController constructor.
+     * @param HashRepositoryInterface $hashRepository
+     * @param UserRepositoryInterface $userRepository
+     * @param MailManager $mailManager
      */
-    public function __construct()
-    {
+    public function __construct(HashRepositoryInterface $hashRepository,
+                                UserRepositoryInterface $userRepository,
+                                MailManager $mailManager) {
         $this->middleware('guest');
+        $this->hashRepository = $hashRepository;
+        $this->userRepository = $userRepository;
```

Comment:
đoạn này em sửa cho anh $userRepository -> $user
$this->userRepository = $userRepository; thành $this->user = $user tương tự vs hash nhé